packages/compiler/src/core/parser.ts

```diff
@@ -2425,7 +2425,20 @@ function createParser(code: string | SourceFile, options: ParseOptions = {}): Pa
     messageId: keyof CompilerDiagnostics["doc-invalid-identifier"]
   ): ParamLikeTag {
     const name = parseDocIdentifier(messageId);
+
+    if (messageId === "param") {
```

Comment:
This feels like a little hacky of a check, comparing on a message code.

is `@param` the only one that allows `-` in tsdoc? I think I would just remove this check and allow it for all `ParamLikeTag` that this function is parsing.

Reply:
Looking at `@typeParam` https://tsdoc.org/pages/tags/typeparam/ seems to be also using hyphen so I would just remove that if and always try to parse `-`

Reply:
Oh! 🤦🏻‍♂️ `typeparam` = `@typeParam`. I had not realized that; yes, I can re-evaluate and share this logic.

I will update today or tomorrow.

Reply:
We have `@template` which is `@typeParam` in tsdoc(or also allows `@template`) and `@param` (same as in tsdoc) both of those are `ParamLikeTag` which have a parameter name and I believe then we can argue could have this optional `-`